libraries/AP_HAL/utility/RingBuffer.cpp

```diff
@@ -6,9 +6,14 @@
   implement a simple ringbuffer of bytes
  */
 
+static inline uint32_t align_power2(uint32_t u)
+{
+    return 1U << ((sizeof(u) * 8) - __builtin_clz(u - 1));
```

Comment:
I didn't get why you have to use u - 1. For sizes that are <some p2 aligned number> + 1 (like 3, 9, 17) the allocated size is 1 byte less of the requested one. Is this expected for any reason?